modules/apps/questions/questions-test/src/testIntegration/java/com/liferay/questions/web/internal/upgrade/v1_1_0/test/QuestionsConfigurationUpgradeProcessTest.java

```diff
@@ -0,0 +1,300 @@
+import com.liferay.portal.kernel.test.rule.AggregateTestRule;
+import com.liferay.portal.kernel.test.util.RandomTestUtil;
+import com.liferay.portal.kernel.test.util.ServiceContextTestUtil;
+import com.liferay.portal.kernel.test.util.TestPropsValues;
+import com.liferay.portal.kernel.upgrade.UpgradeProcess;
+import com.liferay.portal.kernel.upgrade.UpgradeStep;
+import com.liferay.portal.kernel.util.HashMapDictionary;
+import com.liferay.portal.kernel.util.HashMapDictionaryBuilder;
+import com.liferay.portal.test.log.LogCapture;
+import com.liferay.portal.test.log.LoggerTestUtil;
+import com.liferay.portal.test.rule.Inject;
+import com.liferay.portal.test.rule.LiferayIntegrationTestRule;
+import com.liferay.portal.test.rule.PermissionCheckerMethodTestRule;
+import com.liferay.portal.upgrade.registry.UpgradeStepRegistrator;
+
+import java.util.ArrayList;
+import java.util.Dictionary;
+import java.util.List;
+import java.util.Objects;
+
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.ClassRule;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import org.osgi.framework.Constants;
+import org.osgi.service.cm.Configuration;
+import org.osgi.service.cm.ConfigurationAdmin;
+
+/**
+ * @author Marco Galluzzi
+ */
+@RunWith(Arquillian.class)
+public class QuestionsConfigurationUpgradeProcessTest {
+
+	@ClassRule
+	@Rule
+	public static final AggregateTestRule aggregateTestRule =
+		new AggregateTestRule(
+			new LiferayIntegrationTestRule(),
+			PermissionCheckerMethodTestRule.INSTANCE);
+
+	@Before
+	public void setUp() throws Exception {
+		Configuration systemConfiguration = _getSystemConfiguration();
+
+		if (systemConfiguration != null) {
+			_originalSystemConfigurationProperties =
+				systemConfiguration.getProperties();
+
+			systemConfiguration.delete();
+		}
+
+		Configuration[] scopedConfigurations = _getScopedConfigurations();
+
+		_originalScopedConfigurationsProperties = new ArrayList<>();
+
+		for (Configuration scopedConfiguration : scopedConfigurations) {
+			if (scopedConfiguration != null) {
+				_originalScopedConfigurationsProperties.add(
+					scopedConfiguration.getProperties());
+
+				scopedConfiguration.delete();
+			}
+		}
+	}
+
+	@After
+	public void tearDown() throws Exception {
+		_deleteScopedConfigurations();
+		_deleteSystemConfiguration();
+
+		if (_originalSystemConfigurationProperties != null) {
+			_createSystemConfiguration(_originalSystemConfigurationProperties);
+		}
+
+		for (Dictionary<String, Object> properties :
+				_originalScopedConfigurationsProperties) {
+
+			_createScopedConfiguration(properties);
+		}
+	}
+
+	@Test
+	public void testUpgradeWithDefaultTopicRootId() throws Exception {
+		_testUpgrade(
+			StringPool.BLANK, MBCategoryConstants.DEFAULT_PARENT_CATEGORY_ID);
+	}
+
+	@Test
+	public void testUpgradeWithInvalidTopicRootId() throws Exception {
+		_testUpgrade(StringPool.BLANK, 1);
```

Comment:
Instead of hardcoding `1`, can you use a random value > 0? (e.g. `RandomTestUtil.randomLong()`)

Reply:
Well, I see `1` as an always-invalid value, while a random number can be valid if we have bad luck. What do you think?

Reply:
But that shouldn't happen, as you're not creating any entry in that test.

This is not a big deal, but the use of random will make it clear that the value doesn't matter at all (only that it doesn't exist).

Reply:
Done. See last commit [574c892](https://github.com/liferay-content-management/liferay-portal/pull/5927/commits/574c892318435a0934d4a9a824b8ecb6d522ebb8)